### ISSUE TITLE
Form doesn't do "Successful Submit Action", stuck at "Please wait..."

### ISSUE THREAD

**Issue by imihandstand** [mautic/mautic#9909]
## Bug Description
For whatever reason my form stopped redirecting the visitor if he submits the form - and the button stucks at "Please wait...". Even If I use "Display message" for "Successful Submit Action", the same happens: The button says "Please wait..." forever, without showing the message.

However, it add the contact correctly to Mautic, the form sends the email, etc.

I tested it with 2 websites of mine (on 2 different servers).

I originally used Mautic 3.2.5, but updated now to 3.3.2 to hopefully resolve this issue, but no, it still doesn't work.

EDIT: Safari-console shows this error:
"_Refused to display 'https://subdomain.domain.com/mautic/form/submit?formId=10' in a frame because it set 'X-Frame-Options' to 'SAMEORIGIN'._"

| Q   | A
| --- | ---
| Mautic version | 3.2.5 and 3.3.2
| PHP version | 7.3.27
| Browser | Firefox, Opera, Safari

**Comment by imihandstand** [comment]
I think I **temporarily** solved this now by adding to the .htaccess-file this:

_Header always unset X-Frame-Options_

I found this at https://github.com/mautic/mautic/issues/7498 , but I'm not sure if this is an okay solution, plus I might have to do it after every single update of Mautic... So I would still like to have a real solution :)

Thank you!

**Comment by npracht** [comment]
Hello,

Do you host your mautic instance on same domain of your website where the form is embedded ?
Also, don't you have CORS enabled in your mautic instance ?

Thanks

**Comment by imihandstand** [comment]
> Do you host your mautic instance on same domain of your website where the form is embedded ?
No, I have a unique domain/server just for Mautic (and it's installed on a subdomain - like "email.example.com", but I don't use the main domain "example.com" for anything). Forms are in 2-3 other websites with other domains.

I just saw, that on the Mautic-Forum somebody wrote a [workaround with some CNAME trick](https://forum.mautic.org/t/multiple-anonym-contacts-for-same-visitor-even-after-identifying-him/19094/10?u=imihandstand). Might try that out.

> Also, don't you have CORS enabled in your mautic instance ?

No, I always have CORS disabled. 1-2 years ago when I tried out CORS, it somehow didn't work, so it is just disabled. Since than I didn't retry it.
